GameServerLib/API/ApiFunctionManager.cs

```diff
@@ -631,9 +631,22 @@ public static void SetAutocast(ISpell spell)
             spell.SetAutocast();
         }
 
+        /// <summary>
+        /// Sets the state of the given status for the given unit.
+        /// </summary>
+        /// <param name="unit">Unit to set status.</param>
+        /// <param name="status">Status to set.</param>
+        /// <param name="enabled">Whether or not the status should be enabled.</param>
         public static void SetStatus(IAttackableUnit unit, StatusFlags status, bool enabled)
         {
-            unit.SetStatus(status, enabled);
+            // Loop over all possible status flags and set them individually.
+            foreach (StatusFlags enumFlag in Enum.GetValues(typeof(StatusFlags)))
```

Comment:
Enum.GetValues is a very slow function. Consider replacing it with something else.

Reply:
Resolved.